### DISCUSSION TITLE
Welcome to Kener Feature Requests

### DISCUSSION THREAD

**Discussion by rajnandan1** [rajnandan1/kener#119] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Request for a feature
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by astoko** [comment]
Add SMTP Fuctionality. I think its an non-brainer to add that.

**Reply by rajnandan1** [reply]
Yes

**Reply by rajnandan1** [reply]
This is done

**Reply by astoko** [reply]
I saw in the change logs that if you put both it will use one?

Maybe the best way is to set a hierarchy system for fallbacks and options to send to all.

What about multiple email agents?

Throwing ideas at the wall if anything sticks yk?

**Reply by rajnandan1** [reply]
No while adding a trigger you can choose between smtp or resend. 

Where smtp will take priority is when you set in env variable for use cases like forgot password. 

For triggers you can set up multiple smtps

**Comment by MichaelErmer** [comment]
Add functionality to add "scheduled maintenance" windows that will happen in the future, show which service they affect and how log they are planed, as bonus could even add cron based repeating windows.

**Reply by rajnandan1** [reply]
This is done

**Comment by rajnandan1** [comment]
Will add it, also asked in #224

**Comment by MichaelErmer** [comment]
Please add option to disable Google Fonts (that also removes the prefetch headers) or allow upload of fonts to serve then locally, there's some legal privacy issues with google fonts from their cdn.

**Reply by rajnandan1** [reply]
This is done

**Comment by sqkkyzx** [comment]
If I try to add a trigger type to the project, could you please confirm if the following steps are correct?

1. Copy `/src/lib/server/notification/slack.js` as a base template to modify.
2. Reference it in `src/lib/server/notification/notif.js`.
3. Add it in the `<label class="cursor-pointer">` in `src/lib/components/manage/monitorsAdd.svelte`.
4. Include it in `<Card.Title>` and `<div class="flex justify-stretch gap-4">` in `src/lib/components/manage/triggerInfo.svelte`.

Is this sufficient? I apologize for my lack of experience with Node.js project development.

**Reply by sqkkyzx** [reply]
Additionally, I would like to know if I can merge it into the project once I have completed it?

**Reply by rajnandan1** [reply]
May be if you raise a PR I can look at it. But these steps are on the right track. Thanks

**Reply by kaffolder7** [reply]
@sqkkyzx I'm hoping to push a Telegram trigger type soon. I need to complete some testing and add a few screenshots to the docs first, but the bulk of the code is in my [fork](https://github.com/kaffolder7/kener/tree/feature/telegram-trigger). 🏃‍♂️‍➡️

**Reply by rajnandan1** [reply]
I have an example for telegram https://kener.ing/docs/triggers/#examples-telegram

**Reply by kaffolder7** [reply]
Oh geez, silly me! 😅  That's what I get for not fully parsing the documentation. 🤦 I can still submit my PR at some point if you think it would be helpful to abstract Telegram into its own trigger type. Thinking out loud though, I wonder if it would make sense to try implementing a library that handles numerous notification types, like [AppRise](https://github.com/caronc/apprise), etc? Then you wouldn't need to as closely maintain the various trigger endpoint/payload formats, and instead let that library (& community) handle updates.

**Comment by xCrypThor** [comment]
I would like to have a rich-text editor in the incident summary and maintenance summary.
How can this be implemented, or where can I extend it?
Can someone help me or include it in the next version?

**Reply by rajnandan1** [reply]
this is added as a markdown editor v3.2.11

**Comment by dug42** [comment]
Add functionality to process gRPC calls.

**Comment by aba-zwicker** [comment]
I'd like to add a maintenance event via API.
Currently it's only a ui feature. Would it be possible to extend the API?

https://github.com/rajnandan1/kener/blob/1a67560137ad590349edede0e8dcbf7a9a848f03/src/routes/(kener)/api/incident/%2Bserver.js#L20

The current implementation of the incident API ignores the end date, state and status.

```bash
curl 'http://localhost:3000/api/incident' \
  --request POST \
  --header 'Content-Type: application/json' \
  --header 'Authorization: <redacted>' \
  --data '{
  "start_date_time": 1744717097,
  "end_date_time": 1744724289,
  "incident_type": "MAINTENANCE",
  "title": "test api maintenance 1",
  "state": "RESOLVED",
  "status": "OPEN"
}'
{"id":6,"start_date_time":1744717097,"end_date_time":null,"created_at":"2025-04-15 12:33:26","updated_at":"2025-04-15 12:33:26","title":"test api maintenance 1","status":"OPEN","state":"INVESTIGATING"}
```

**Reply by rajnandan1** [reply]
Will add it

**Comment by dKenGuru** [comment]
1. Allow edit "Site Name" width. There a lot space, but ~18 digits start write in two rows.
2. Monitoring latency - lowest, highest, average.
3. Maybe add option to set and choose country flags for monitor name. Built-in. I know about image/logo option.
4. Triggers via Telegram

**Reply by rajnandan1** [reply]
There is a telegram example https://kener.ing/docs/triggers#webhook-examples

**Reply by vedmaka** [reply]
+1 for latency monitoring

**Comment by ngironi** [comment]
Hello

Thanks for your work, it's very nice and simple. 

I have 2 requests/ideas :
- Is it possible to add a customization of the template emails for event_update or email_code ? (I would love being able to translate them)
- Add capability to "force" some subscribers email ? 

Thanks

**Reply by rajnandan1** [reply]
Wil add it in v4
